src/domain/blockchain/usecase/SendResponseToOracleUseCase.js

```diff
@@ -0,0 +1,18 @@
+class SendResponseToOracleUseCase {
+  constructor(oracle, logger) {
+    this.oracle = oracle;
+    this.logger = logger;
+  }
+
+  async sendResponse(response) {
+    try {
+      await this.oracle.sendResponse(response);
+      this.logger.info(`Response sent to blockchain  [requestId=${response.requestId}]`);
+    } catch (e) {
+      this.logger.error(`Failed to send response to blockchain [requestId=${response.requestId}]`);
+      throw e;
```

Comment:
Either log or throw - https://stackoverflow.com/questions/6639963/why-is-log-and-throw-considered-an-anti-pattern

Reply:
Doing so this way would abstract this use case. Firstly the use case know and log what happens. Secondly the meta use case (ExecutereadyRequestsUseCase) make decision dependent on success or error in base use case. What we can do is to throw simple error instead of bubble this one up.